File: lucky7_java/src/com/game/luckySeven/objects/Slot.java
package com.game.luckySeven.objects;

import com.game.luckySeven.framework.BufferedImageLoader;
import com.game.luckySeven.framework.Texture;

import java.awt.*;
import java.awt.image.BufferedImage;

public class Slot {

	private int x, y;
	private int width, height;
	private int value;
	private Texture tex;
	private BufferedImageLoader loader;
	private BufferedImage image;


	public Slot(int x, int y, int width, int height) {
		this.x = x;
		this.y = y;
		this.width = width;
		this.height = height;

		tex = new Texture();
	}

	public void tick() {

	}

	public void render(Graphics g) {
		g.setColor(Color.gray);
		g.drawRect(x, y, width, height);

		if (value == 0) g.drawImage(tex.slot_pirate, x, y, width, height, null);
		else if (value == 7) g.drawImage(tex.slot_seven, x, y, width, height, null);
		else g.drawImage(tex.slot_coin, x, y, width, height, null);

	}

	public int getX() {
		return x;
	}

	public void setX(int x) {
		this.x = x;
	}

	public int getY() {
		return y;
	}

	public void setY(int y) {
		this.y = y;
	}

	public int getWidth() {
		return width;
	}

	public void setWidth(int width) {
		this.width = width;
	}

	public int getHeight() {
		return height;
	}

	public void setHeight(int height) {
		this.height = height;
	}

	public int getValue() {
		return value;
	}

	public void setValue(int value) {
		this.value = value;
	}
}

File: lucky7_javascript/src/slot.js
export default class Slot {


  constructor(x, y, width, height, canvas) {
    this.x = x;
    this.y = y;
    this.width = width;
    this.height = height;
    this.canvas = canvas;

    this.value = 0;
    this.path = "";
  }

  draw(ctx) {
    
    const xa = this.x;
    const ya = this.y;
    const wa = this.width;
    const ha = this.height;

    if (this.value == 0) this.path = "res/slot_pirate.png";
    else if (this.value == 7) this.path = "res/slot_seven.png";
    else this.path = "res/slot_coin.png";

    var imgSlot = new Image(this.width, this.height);

    imgSlot.onload = function () {
      ctx.drawImage(imgSlot, xa, ya, wa, ha);
     }
    imgSlot.src = this.path;

    ctx.strokeRect(this.x - 5, this.y - 5, this.width + 10, this.height + 10);
  }

  
setValue(value) {
  this.value = value;
}

getValue() {
  return this.value;
}

}